frcxx_nft/src/state.rs

```diff
@@ -171,109 +221,1284 @@ impl NFTState {
         Ok(balance)
     }
 
-    /// Burns a token, removing it from circulation and deleting associated metadata
-    pub fn burn_token<BS: Blockstore>(&mut self, bs: &BS, token_id: TokenID) -> Result<()> {
+    /// Approves an operator to transfer a set of specified tokens
+    ///
+    /// Checks that the caller is the owner of the specified token. If any of the token_ids is not
```

Comment:
Also this is too much logic for the state object. Business logic belongs in the token library, so that someone could use this state directly to achieve other flows. The state should maintain invariants, but not implement policy – it should be dumb. Ditto for some methods below.

This might conflict with batching somewhat, where we want per-token logic implemented in the token lib caller. Maybe a callback pattern or richer return value can help.

Reply:
https://github.com/helix-onchain/filecoin/issues/157

Reply:
https://github.com/helix-onchain/filecoin/issues/162